### ISSUE TITLE
Shorter IDs for human readability 

### ISSUE THREAD

**Issue by wesort** [mariovass/statamic-uuid#12]
I'm curious whether a shorter unique/random ID could be generated? 

I'm building a site for a film and image archive and we need to auto-generate a human readable ID (6 or 8 characters, A-Z 0-9). There will be around 10k items in the collection so in practice the uniqueness probability for a shorter string _should_ be fine. 

One option I've considered is to truncate the {s} `id` to first 8 characters but I'm not sure how I'd present that in the CP as readonly and truncated. 

Thanks.

**Comment by aryehraber** [comment]
Hey @wesort! Interesting idea/use-case, not had this request before.

I think we could provide a "custom alphabet" & "size" option, which switches the underlying library from [uuidv4](https://github.com/thenativeweb/uuidv4) to [nanoid](https://github.com/ai/nanoid) which has the ability for such customisations: https://github.com/ai/nanoid/#custom-alphabet-or-size

Of course the likelihood of a collision is much higher with the short string and fairly limited alphabet that you've proposed, but it sounds like you're already aware of that.

How does that sound to you?

**Comment by wesort** [comment]
That sounds incredible. 

Since my previous comment I’ve seen how having a shorter ID might be useful for another project too where the client needs SKUs for their products.

Can I do anything to help? I don’t really have any JS skills I’m afraid.

**Comment by aryehraber** [comment]
I'll try and hack on this over the weekend and ping you when it's ready for testing!

**Comment by aryehraber** [comment]
Just released `v2.1.0` which contains a few new fieldtype options 🎉  

See usage and example in [README](https://github.com/aryehraber/statamic-uuid#usage) for more details, but in terms of your above mentioned requirements, I think your config would be:

```yaml
fields:
  -
    handle: my_short_id
    field:
      type: uuid
      id_type: nanoid
      alphabet: ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789
      size: 8
```

Let me know if you have any issues/questions!
